tableaudocumentapi/archivefile.py

```diff
@@ -0,0 +1,67 @@
+import contextlib
+import os
+import shutil
+import tempfile
+import zipfile
+
+import xml.etree.ElementTree as ET
+
+
+@contextlib.contextmanager
+def temporary_directory(*args, **kwargs):
+    d = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+def find_file_in_zip(zip, ext):
+    for filename in zip.namelist():
+        if os.path.splitext(filename)[-1].lower() == ext[:-1]:
+            return filename
+
+
+def get_xml_from_archive(filename):
+    file_type = os.path.splitext(filename)[-1].lower()
+    with temporary_directory() as temp:
+        with zipfile.ZipFile(filename) as zf:
```

Comment:
Since we reextract this when saving the changes, why not move this to use zipfile.open on the archive file directly, avoiding having multiple tempdirs and not needing to extract the extract (oi) unless the user is saving the file.